### ISSUE TITLE
Allow `turso dev` to persist to local file

### ISSUE THREAD

**Issue by arjunyel** [chiselstrike/turso-cli#418]
In some environments such as Cloudflare Workers you can't use `file:` URLs because they don't let you use the filesystem.

Instead, we could connect to `turso dev` running locally and have an option to persist to a local LibSQL file such as `file:somefile.db`. This would be an optional alternative to the current in-memory support.

Fixes https://github.com/libsql/libsql-client-ts/issues/21

**Comment by CodingDoug** [comment]
There are also a number of other ways to start a local sqld, which you can read about in the docs for sqld:

https://github.com/libsql/sqld/blob/main/docs/BUILD-RUN.md

They all persist data in a file by default and you can connect to them using a localhost URL.  You can also work with the database file using the sqlite3 command line.

**Comment by arjunyel** [comment]
Sorry, what's the difference between turso dev and running sqld?

**Comment by glommer** [comment]
you do have the option of running a local file with `turso dev` by passing the option `--local-file`. But as Doug mentioned, it is experimental and won't support interactive transactions.

We are moving interactive transactions over pure HTTP instead of WebSockets, so we'll not touch `turso dev` until that gets done.

**Comment by ben-xD** [comment]
It seemed important to run migrations on the local database, to set up the data? I launched the in-memory database with `turso dev`, then tried `turso db shell http://localhost:3030`, but commands like `select * from users`; gave an error:
```
Error: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub
```

If the database was persisted, I suspect it would be pretty simple to edit it. I'll take a look at sqld, thanks!

---

My application can't use the file directly, given that it runs in Cloudflare Workers.

**Comment by ben-xD** [comment]
When I switched from `import { createClient } from '@libsql/client/web';` to `import { createClient } from '@libsql/client/http';`, I got another error:

`@libsql/client` tries to use `https` and the worker fails:
```
[0] workerd/jsg/util.c++:276: error: e = kj/compat/tls.c++:74: failed: OpenSSL error; message = error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER
[0] stack: 1027b92ab 1027bbb1f 1027bc23b 1027bc968 1027c47f8 103760d3c 103762998 1028b3304 1028b3728 1028b2eec 1037506d0 1024495d0 1024468d4 10251c2fc 10245df10 1
```

**Comment by glommer** [comment]
@ben-xD we'll look into some of those issues - thanks! But Cloudflare workers unfortunately may not work with a local IP anyway - this has been reported on our discord. For now the best will be to just create a new database for development in the actual service.

**Comment by CodingDoug** [comment]
Persistence to a local file should be resolved in the latest CLI 0.62. You can pass a `--db-file` argument with a file path to `turso dev`, and it will use that SQLite file to persist data between executions.
